### ISSUE TITLE
[EPIC 2] Core Statistical Modeling Engine

### ISSUE THREAD

**Issue by LinoGoncalves** [LinoGoncalves/GoalCast#2]
## Epic: Core Statistical Modeling Engine

**Business Value:**
As a prediction engine user
I need multiple statistical models (Poisson, Dixon-Coles, Bayesian, xG, copula)
So that I have accurate probability distributions for match outcomes

### Success Metrics
- ✅ Brier score < 0.15 (better than bookmaker baseline)
- ✅ Outcome accuracy +5% vs bookmaker implied probabilities
- ✅ Exact score accuracy ≥ 15%
- ✅ Calibration curves show well-calibrated predictions
- ✅ xG integration improves model lift by measurable margin

### User Stories (Sub-Issues to be Created)
1. Implement baseline Poisson model
2. Implement Dixon-Coles adjustment for low scores
3. Implement Bayesian hierarchical team strength estimation
4. Integrate xG data to adjust attack/defense parameters
5. Implement copula-based joint score distributions
6. Create model calibration framework (Brier, log loss)
7. Build model comparison and selection logic

### Technical Approach
- **Tech Stack**: Python 3.13+, scipy, statsmodels, scikit-learn, PyTorch (optional for advanced models)
- **Bayesian**: PyMC for hierarchical modeling
- **Copula**: scipy.stats for joint distributions
- **Validation**: Cross-validation with historical seasons
- **Output**: Structured JSON per API contract (`probs_model.json`)

### Acceptance Criteria (High-Level)
```gherkin
Given historical fixtures, odds, and xG data
When the statistical models run
Then probability distributions are generated for all outcomes (H/D/A)
And exact scoreline probabilities are provided (0-0 through 5-5+)
And model calibration metrics are calculated vs bookmaker baseline
And outputs are in structured JSON format per API contract
And Brier score < 0.15 is achieved
```

### Dependencies
- **Depends on**: #1 (Data Foundation & Ingestion Pipeline)

### Priority
**P0 - Critical**: Core prediction capability

### Complexity Estimate
**XL - Very Complex (>4 weeks)**: Multiple advanced models, Bayesian inference, xG integration, calibration
